sdk/Trace/TraceContextPropagator.php

```diff
@@ -16,40 +22,55 @@
  * traceparent header and relevant parts of the tracestate header containing
  * their proprietary information.
  */
-final class TraceContext implements API\TextMapFormatPropagator
+final class TraceContextPropagator implements API\TextMapPropagator
 {
     public const TRACEPARENT = 'traceparent';
     public const TRACESTATE = 'tracestate';
-    private const VERSION = '00'; // Currently only '00' is supported
+    private const VERSION = '00'; // Currently, only '00' is supported
 
-    /**
-     * {@inheritdoc}
-     */
+    /** {@inheritdoc} */
     public static function fields(): array
     {
         return [self::TRACEPARENT, self::TRACESTATE];
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public static function inject(API\SpanContext $context, &$carrier, API\PropagationSetter $setter): void
+    /** {@inheritdoc} */
+    public static function inject(&$carrier, Context $context = null, PropagationSetter $setter = null): void
     {
+        $context = $context ?? Context::getCurrent();
+        $setter = $setter ?? new PropagationMap();
```

Comment:
Need to read the spec a bit more, but might be better to have these be singletons and expose them via class methods.

Reply:
I think you're correct here - I can't think of cases where we'd have more than one setter.